src/rascal/structure_managers/adaptor_strict.hh

```diff
@@ -225,12 +227,15 @@ namespace rascal {
       return this->manager->get_shared_ptr();
     }
 
-    // BUG8486@(till) I deleted the non const getters, because they are not
-    // needed
-    // if this was wrong, please explain
+    //! Get the manager used to build the instance
+    ConstImplementationPtr_t get_previous_manager_impl() const {
+      return this->manager->get_shared_ptr();
+    }
+
     //! returns the distance between atoms in a given pair
     template <size_t Order, size_t Layer>
-    double get_distance(const ClusterRefKey<Order, Layer> & pair) const {
+    const double &
```

Comment:
> Btw: we changed distance property a little bit on our side: it is now registered with the manager. Does not change the interface to the distances here. Just saying...

This is not a problem to me since it is already the case: a shared pointer in adaptor_strict and another one in the manager's map.